app/models/service_provider.rb

```diff
@@ -158,15 +158,25 @@ def certs_are_pems
   end
 
   def validate_attribute_bundle
-    if (attribute_bundle || []).empty?
+    # attribute bundle should not be empty when saml and ial2 are selected
+    if !attribute_bundle.present? && ial == 2 && identity_protocol == 'saml'
       errors.add(:attribute_bundle, 'Attribute bundle cannot be empty')
       return false
     end
 
-    possible_attributes = ALLOWED_IAL1_ATTRIBUTES + ALLOWED_IAL2_ATTRIBUTES
-    return if ial == 2 && (attribute_bundle - possible_attributes).empty?
-    return if (attribute_bundle - ALLOWED_IAL1_ATTRIBUTES).empty?
+    if attribute_bundle.present?  && contains_invalid_attribute?
```

Comment:
```suggestion
    if attribute_bundle.present? && contains_invalid_attribute?
```
extra space snuck in there

Reply:
Ah caught it before I did!

Reply:
I've become very dependent on linters...